src/tools/blast/blastcommandline.jl

```diff
@@ -0,0 +1,49 @@
+# wrapper for BLAST+ command line functions
+
+"""
+`blastn(query, subject, flags...)``
+Runs blastn on `query` against `subject`. Query may be string or DNASequence.
+    Subjects and queries may be file names (as strings), DNASequence type or
+    Array of DNASequence.
+    May include optional `flags` such as `[-pident, 95]`. Do not use `-outfmt`.
+"""
+function blastn(query::ASCIIString, subject::ASCIIString, flags...)
+    result = readall(`blastn -query $query -subject $subject -outfmt 6 $flags`)
+    # do stuff
+end
+
+function blastn(query::DNASequence, subject::DNASequence, flags...)
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    result = readall(`blastn -query $querypath -subject $subjectpath -outfmt 6 $flags`)
```

Comment:
I think it's ok. I've done similar things before in method dispatch. So long as people read the docs this is pretty non-confusing. You define a method blast, which works on DNASequence, it achieves it function, by writing the DNASequence objects to temp files, and it passes the paths, to the blasts method that is designed to work on filepaths. Indeed defining a method of blasts for use with file paths, and then not using it in other methods seems like a case of the method to me.